src/typechecker/FStar.TypeChecker.Rel.fs

```diff
@@ -4278,8 +4278,14 @@ let discharge_guard' use_env_range_msg env (g:guard_t) (use_smt:bool) : option<g
                     Options.with_saved_options (fun () ->
                         ignore <| Options.set_options "--no_tactics";
                         let vcs = env.solver.preprocess env vc in
-                        vcs |> List.map (fun (env, goal, opts) ->
-                        env, norm_with_steps "FStar.TypeChecker.Rel.norm_with_steps.7" [Env.Simplify; Env.Primops] env goal, opts)
+                        let vcs = List.map (fun (env, goal, opts) ->
+                        env, norm_with_steps "FStar.TypeChecker.Rel.norm_with_steps.7" [Env.Simplify; Env.Primops] env goal, opts) vcs in
```

Comment:
I wondered about this too. But I figured Aymeric wanted to intercept the goals exactly as they are before the SMT encoding sees them.

OTOH, it would be easy for the handler to call norm [simplify;primops] itself.

Reply:
Indeed, my idea was to be able to more easily debug/handle queries exactly as passed to the SMT. I agree it would be easy to call norm from the tactic as a user, but this would require expert knowledge that these specific normalization steps occur normally when encoding to SMT

Reply:
I would expect the normalization to still happen, hence making
```
[@handle_smt_goals]
let tau : Tac unit = ()
```
a no-op, instead of potentially causing the SMT to try to solve a very different query.

Reply:
In today's F* meeting, we agreed that having handle_smt_goals before this small normalization wouldn't bring much, and that having goals exactly as passed to SMT was more valuable. We also discussed the possibility of exposing a hook to get the goal before any kind of normalization and preprocessing, but decided that this was an orthogonal feature that, if needed, could be added in a subsequent PR.